src/components/Tab/Tab.tsx

```diff
@@ -0,0 +1,149 @@
+/** @jsxImportSource theme-ui */
+import { get } from "@theme-ui/css"
+import React, { useCallback, useState } from "react"
+import { useTabContext } from "./context"
+import { TabContext } from "./context"
+
+export interface TabProps {
+  initialOpenIndex?: number
+  children: React.ReactNode
+  activeBackgroundColor?:
+    | "primary"
+    | "secondary"
+    | "text"
+    | "success"
+    | "warning"
+}
+
+const Tab = ({
+  children,
+  initialOpenIndex,
+  activeBackgroundColor = "secondary",
+}: TabProps) => {
+  const [openIndex, setOpenIndex] = useState<number | undefined>(
+    initialOpenIndex
+  )
+
+  const toggleOpen = useCallback(
+    (index: number) => {
+      if (openIndex === index) setOpenIndex(undefined)
+      else setOpenIndex(index)
+    },
+    [openIndex]
+  )
+
+  const allItems = React.Children.toArray(children)
+    .filter((child) => {
+      return React.isValidElement(child)
+    })
+    .map((child, index) => (
+      <TabContext.Provider
+        value={{ index, toggleOpen, openIndex, activeBackgroundColor }}
+        key={index}
+      >
+        {child}
+      </TabContext.Provider>
+    ))
+
+  const tabLink = allItems.filter((item) =>
+    item.props.children.props.hasOwnProperty("title")
+  )
+  const tabPanel = allItems.filter((item) =>
+    item.props.children.props.hasOwnProperty("index")
+  )
```

Comment:
You could use the `TabContext.Provider` in these arrays instead of the general `allItems` array, that way you could use the context index in the `Tab.Panels`, and not have it as a prop, which makes it even simpler for the user.
You would just need a different way of identifying which children is a `Link` or a `Panel`, as the panel wouldn't have an `index` prop.